include/net/ieee802154_radio.h

```diff
@@ -71,15 +71,6 @@ struct ieee802154_radio_api {
 			  enum ieee802154_filter_type type,
```

Comment:
This could also be merged with "drivers/ieee802154: Stop exposing old radio API functions" as this commit looks a bit redundant now. In that case the commit subject could be changed "drivers/ieee802154: Remove old radio API functions".

Reply:
I won't merge, as the other patch touches drivers and this one the public API in include. I like to keep things per-domain as much as possible (it's not always possible but here it is.).